storage/imxrt-flash/flashdrv.c

```diff
@@ -15,141 +15,50 @@
-	xfer.seqNum = 1;
-	xfer.isParallelModeEnable = 0;
-
-	return -flexspi_norFlashExecuteSeq(ctx->instance, &xfer);
-}
-
-
-static int flash_writeBytes(flash_context_t *ctx, uint32_t dstAddr, uint32_t *src, uint32_t size)
-{
-	flexspi_xfer_t xfer;
-
-	xfer.txSize = size;
-	xfer.txBuffer = src;
-	xfer.baseAddress = dstAddr;
-	xfer.operation = kFlexSpiOperation_Write;
-	xfer.seqId = PAGE_PROGRAM_SEQ_ID;
-	xfer.seqNum = 1;
-
-	return -flexspi_norFlashExecuteSeq(ctx->instance, &xfer);
-}
-
-
-static int flash_waitBusBusy(flash_context_t *ctx)
-{
-	int err = EOK;
-	uint32_t buff;
-	flexspi_xfer_t xfer;
-	uint8_t retrans = 0;
-	const uint8_t MAX_RETRANS = 15;
-
-	xfer.rxSize = 4;
-	xfer.rxBuffer = &buff;
-	xfer.baseAddress = ctx->instance;
-	xfer.operation = kFlexSpiOperation_Read;
-	xfer.seqId = READ_STATUS_REG_SEQ_ID;
-	xfer.seqNum = 1;
-	xfer.isParallelModeEnable = 0;
-
-	do {
-		usleep(retrans * 500);
-		err = flexspi_norFlashExecuteSeq(ctx->instance, &xfer);
-	} while ((buff & 0x3) && (++retrans < MAX_RETRANS) && (err == 0));
-
-	return err;
-}
-
-
-static int flash_getVendorID(flash_context_t *ctx, uint32_t *manID)
-{
-	flexspi_xfer_t xfer;
-
-	/* READ JEDEC ID*/
-	ctx->config.mem.lut[4 * READ_JEDEC_ID_SEQ_ID] = LUT_INSTR(LUT_CMD_CMD, LUT_PAD1, FLASH_SPANSION_CMD_RDID, LUT_CMD_READ, LUT_PAD1, FLASH_SPANSION_CMD_WRDI); // 0x2404049f
-	ctx->config.mem.lut[4 * READ_JEDEC_ID_SEQ_ID + 1] = 0;
-	ctx->config.mem.lut[4 * READ_JEDEC_ID_SEQ_ID + 2] = 0;
-	ctx->config.mem.lut[4 * READ_JEDEC_ID_SEQ_ID + 3] = 0;
-	flexspi_norFlashUpdateLUT(ctx->instance, READ_JEDEC_ID_SEQ_ID, (const uint32_t *)&ctx->config.mem.lut[4 * READ_JEDEC_ID_SEQ_ID], 1);
-
-	xfer.rxSize = 3;
-	xfer.rxBuffer = manID;
-	xfer.baseAddress = ctx->instance;
-	xfer.operation = kFlexSpiOperation_Read;
-	xfer.seqId = READ_JEDEC_ID_SEQ_ID;
-	xfer.seqNum = 1;
-	xfer.isParallelModeEnable = 0;
-
-	return flexspi_norFlashExecuteSeq(ctx->instance, &xfer);
-}
-
-
-/* Read/write/erase operation via ROM API */
-
-static int flash_isValidAddress(flash_context_t *context, uint32_t addr, size_t size)
+static int flash_isValidAddress(flash_context_t *context, addr_t addr, size_t size)
 {
-	if ((addr + size) <= context->properties.size)
+	if ((addr + size) <= context->properties.size) {
 		return 0;
+	}
 
 	return 1;
 }
 
 
-ssize_t flash_readData(flash_context_t *ctx, uint32_t offset, char *buff, size_t size)
+ssize_t flash_readData(flash_context_t *ctx, uint32_t offset, void *buff, size_t size)
 {
-	if (flash_isValidAddress(ctx, offset, size))
-		return -1;
-
-	if (flexspi_norFlashRead(ctx->instance, &ctx->config, buff, offset, size) < 0)
+	if (flash_isValidAddress(ctx, offset, size)) {
 		return -1;
+	}
 
-	return size;
+	return nor_readData(&ctx->fspi, ctx->port, offset, buff, size, 0);
 }
 
 
-ssize_t flash_directBytesWrite(flash_context_t *ctx, uint32_t offset, const char *buff, size_t size)
+ssize_t flash_directBytesWrite(flash_context_t *ctx, uint32_t offset, const void *buff, size_t size)
```

Comment:
Doesn't really make sense to use `addr_t` - it is a type holding physical address on our target machine, it's size does not correspond in any way with address space of NOR memory - it describes our target computer. So e.g. on 8-bit machine it would hold 16 bits.

uint32_t is ok, because NOR memories have max address space of 4 bytes. Imho we shouldn't use any standard type for this (uintptr_t, addr_t, off_t), because these are related to the CPU rather than NOR peripheral component.